presto-main/src/main/java/com/facebook/presto/type/CharOperators.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.type;
+
+import com.facebook.presto.spi.function.LiteralParameters;
+import com.facebook.presto.spi.function.ScalarOperator;
+import com.facebook.presto.spi.function.SqlType;
+import com.facebook.presto.spi.type.StandardTypes;
+import io.airlift.slice.Slice;
+import io.airlift.slice.XxHash64;
+
+import static com.facebook.presto.spi.function.OperatorType.BETWEEN;
+import static com.facebook.presto.spi.function.OperatorType.EQUAL;
+import static com.facebook.presto.spi.function.OperatorType.GREATER_THAN;
+import static com.facebook.presto.spi.function.OperatorType.GREATER_THAN_OR_EQUAL;
+import static com.facebook.presto.spi.function.OperatorType.HASH_CODE;
+import static com.facebook.presto.spi.function.OperatorType.LESS_THAN;
+import static com.facebook.presto.spi.function.OperatorType.LESS_THAN_OR_EQUAL;
+import static com.facebook.presto.spi.function.OperatorType.NOT_EQUAL;
+
+public final class CharOperators
+{
+    private CharOperators() {}
+
+    @LiteralParameters({"x"})
+    @ScalarOperator(EQUAL)
+    @SqlType(StandardTypes.BOOLEAN)
+    public static boolean equal(@SqlType("char(x)") Slice left, @SqlType("char(x)") Slice right)
+    {
+        return left.equals(right);
+    }
+
+    @LiteralParameters({"x"})
+    @ScalarOperator(NOT_EQUAL)
+    @SqlType(StandardTypes.BOOLEAN)
+    public static boolean notEqual(@SqlType("char(x)") Slice left, @SqlType("char(x)") Slice right)
+    {
+        return !left.equals(right);
+    }
+
+    @LiteralParameters({"x", "y"})
```

Comment:
I will make them all have same type. Comparison would work, since `CHAR` operands with different length would be first coerced to `VARCHAR`. However, this is be sub optimal (since chars need to be padded). It also seems strange to have comparison of same length `CHARs` go through one, direct code path and comparison of different length `CHARS` go through coercion to `VARCHAR`. I would prefer to remove `CHAR` operators completely, but I guess it might blow in some other part of system.